frame/bonded-finance/src/tests.rs

```diff
@@ -135,6 +273,46 @@ proptest! {
 
 					  Ok(())
 			  })?;
+	  }
+
+	  #[test]
+	  fn invalid_parts(offer in simple_offer(1)) {
+			  ExtBuilder::build().execute_with(|| {
+						prop_assert_ok!(Tokens::mint_into(NATIVE_CURRENCY_ID, &ALICE, Stake::get()));
+						prop_assert_ok!(Tokens::mint_into(offer.reward_asset, &ALICE, offer.reward_amount));
+						let offer_id = BondedFinance::do_offer(&ALICE, offer.clone());
+					  prop_assert_ok!(offer_id);
+					  let offer_id = offer_id.expect("impossible; qed");
 
+					  prop_assert_ok!(Tokens::mint_into(offer.asset, &BOB, offer.total_price().expect("impossible; qed;")));
+					  prop_assert_eq!(
+							  BondedFinance::bond(Origin::signed(BOB), offer_id, offer.parts + 1),
+							  Err(Error::<Runtime>::InvalidParts.into())
+					  );
+					  prop_assert_eq!(
+							  BondedFinance::bond(Origin::signed(BOB), offer_id, 0),
+							  Err(Error::<Runtime>::InvalidParts.into())
+					  );
+					  Ok(())
+			  })?;
+	  }
+
+	  #[test]
+	  fn offer_completed(offer in simple_offer(1)) {
+			  ExtBuilder::build().execute_with(|| {
+						prop_assert_ok!(Tokens::mint_into(NATIVE_CURRENCY_ID, &ALICE, Stake::get()));
+						prop_assert_ok!(Tokens::mint_into(offer.reward_asset, &ALICE, offer.reward_amount));
+						let offer_id = BondedFinance::do_offer(&ALICE, offer.clone());
+					  prop_assert_ok!(offer_id);
```

Comment:
Rustfmt is completely broken inside macros :/

Reply:
Ah I see ;/ Was about to make a comment about that lol. Yeah because macro content != rust